### PR TITLE
Output configuration for logger

### LINKED ISSUE #38: Use a logging library as a logger implementation instead of console.log
**Is your feature request related to a problem? Please describe.**
I feel that using something like `pino` or `winston` will be best for logging purposes, since there is no need to create our own logger and reinvent the wheel.

**Describe the solution you'd like**
Rewrite the current logger package to use a fully-featured logging library.

**Describe alternatives you've considered**
Using `console.log` has disadvantages, some users would like to save the logs to a file, log via http, etc... This is not possible using our current logging solution.

### PR DESCRIPTION
**Pull request by Arnovsky**
closes #38 

* option to provide output function.
* option to chose if to color the message (useful when logging to a file)
* changed `||` to `??` in the logger package

### CODE REVIEW

File: packages/logger/README.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ Returns the middleware to log HTTP requests.
 
 - `methods` - a list of HTTP methods to log. Defaults to `http`'s `METHODS`
 - `timestamp.format` - timestamp format. Gets consumed by [dayjs](https://day.js.org) library. If a string specified, used as a format, otherwise just eables it.
+- `outputConfiguartion.output` - function that recieved the log generated by the logger package
+- `outputConfiguartion.color` - property that determines if the logger will generate a message with color. useful for logging into the console, disable it if logging into file / http / etc...
```
Comment (talentlessguy) [inline packages/logger/README.md:28]: you have a typo in README

File: packages/logger/README.md [hunk 2]
```diff
@@ -34,7 +36,7 @@ import { logger } from '@tinyhttp/logger'
 const app = new App()
 
 app
-  .use(logger({ methods: ['GET', 'POST'], timestamp: { format: 'HH:mm:ss' } }))
+  .use(logger({ methods: ['GET', 'POST'], timestamp: { format: 'HH:mm:ss' }, outputConfiguration: { output: console.log, color: false } }))
```
Comment (talentlessguy) [inline packages/logger/README.md:39]: What about changing `outputConfiguration` to `output`, and `output` to `function`?

File: packages/logger/src/index.ts [hunk 3]
```diff
@@ -4,16 +4,21 @@ import { IncomingMessage as Request, ServerResponse as Response, METHODS } from
 
 export type LoggerProperties = Partial<{
   methods: string[]
+  outputConfiguration: {
```
Comment (talentlessguy) [inline packages/logger/src/index.ts:7]: Every property in the config must be optional, here `color` and `output` seem to be both required.

EIther put `?` in the end of properties names or use `DeepPartial` ([ref](https://stackoverflow.com/a/49936686))

File: packages/logger/README.md (hunk 2 above)
Comment (Arnovsky) [reply packages/logger/README.md:39]: I thought `callback` might me better, but I think both are ambiguous.

Comment (talentlessguy) [reply packages/logger/README.md:39]: @Arnovsky hmm `callback` sounds good

File: packages/logger/src/index.ts (hunk 3 above)
Comment (Arnovsky) [reply packages/logger/src/index.ts:7]: Wouldn't this mean changing `LoggerProperties` to an interface? since there is no meaning when using `Partial` combined with `?`

Comment (talentlessguy) [reply packages/logger/src/index.ts]: @Arnovsky hmm then probably u're right, it makes more sense to change it to interface.

Also, please cover ur new functionality in tests (if there are none yet) :)

File: packages/logger/src/index.ts [hunk 4]
```diff
@@ -2,18 +2,19 @@ import colors from 'colors'
...
 
-export type LoggerProperties = Partial<{
-  methods: string[]
-  timestamp:
-    | {
-        format?: string
-      }
-    | boolean
-}>
+interface LoggerProperties {
```
Comment (talentlessguy) [inline packages/logger/src/index.ts:5]: Could you please export the interface, and rename it to `LoggerOptions`?

**Comment by talentlessguy** [comment]
@all-contributors please add Arnovsky for code and plugin
